### PR TITLE
HUD implementation

### PR DESCRIPTION
**Pull request by shifatkhan**
The layout will be changed later. This PR is more for the actual functionality of the HUD UI.
image

Check under Assets > SceneSandbox > Shifat > HUDScene

### CODE REVIEW

File: Assets/Scripts/Scoring/PlayerScoreUI.cs [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
 
-public class PlayerUiScore : MonoBehaviour
+public class PlayerScoreUI : MonoBehaviour
 {
-    // TODO can be used for individual player ui
+    [SerializeField] private Team team;
+    public Team Team { get => this.team; }
+    
+    [SerializeField] private TextMeshProUGUI playerNameText;
+    [SerializeField]private TextMeshProUGUI playerScoreText;
```
Comment (eli-lam) [inline Assets/Scripts/Scoring/PlayerScoreUI.cs:13]: Space

File: Assets/Scripts/Scoring/PlayerScoreUI.cs [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+
+    public void UpdateName(string playerName)
+    {
```
Comment (eli-lam) [inline Assets/Scripts/Scoring/PlayerScoreUI.cs:16]: Do ` =>` notation

File: Assets/Scripts/Scoring/PlayerScoreUI.cs [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+        this.playerNameText.text = playerName;
+    }
+    
+    public void UpdateScore(int score)
```
Comment (eli-lam) [inline Assets/Scripts/Scoring/PlayerScoreUI.cs:20]: Do `=>` notation

**Review by eli-lam — APPROVED** [review]

**Comment by eli-lam** [comment]
Also small bug that should be quick to fix, when the timer ends this text shows
image
It would be better to reset to 00:00

File: Assets/Scripts/Scoring/TeamScoreUI.cs [hunk 4]
```diff
@@ -8,15 +9,15 @@ public class TeamScoreUI : MonoBehaviour
     [SerializeField] private Team team;
     public Team Team { get => this.team; }
 
-    private Text text;
+    private TextMeshProUGUI text;
 
     private void Awake()
     {
-        this.text = GetComponent<Text>();
+        this.text = GetComponent<TextMeshProUGUI>();
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Scoring/TeamScoreUI.cs:16]: Since this will be on the prefab, might be more consistent to use the [SerializeField] in the inspector!

Comment (danielwiktorczyk) [reply Assets/Scripts/Scoring/TeamScoreUI.cs:16]: As is in the PlayerScoreUI ^^

**Review by danielwiktorczyk — APPROVED** [review]
Went over in person, looks great! One comment, you might want to add. Otherwise, Squash n Rebase when you're ready!

**Pull request merged**